conda_smithy/lint_recipe.py

```diff
@@ -931,6 +931,31 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
             if msg not in hints:
                 hints.append(msg)
 
+    # 6: Build against on libjpeg-turbo instead of jpeg
+    def hint_jpeg(requirements):
+        if requirements is None:
+            return
+        for req in requirements:
+            if req == "jpeg" or req.startswith("jpeg "):
+                msg = (
+                    "Recipes should usually depend on `libjpeg-turbo` as opposed to "
+                    "`jpeg` for improved performance. For more information please see"
+                    "https://github.com/conda-forge/conda-forge.github.io/issues/673"
+                )
+                if msg not in hints:
+                    hints.append(msg)
+
+    hint_jpeg(requirements_section.get("build", []))
+    hint_jpeg(requirements_section.get("host", []))
+    hint_jpeg(requirements_section.get("run", []))
+    for out in outputs_section:
```

Comment:
`requirements` can be a list instead of a dictionary. Lint in that case too?

Reply:
done.